mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -105,6 +108,15 @@ class GaussianMixtureModel private[ml] (
   def gaussians: Array[MultivariateGaussian] = parentModel.gaussians
 
   @Since("2.0.0")
```

Comment:
I'd prefer to have a wrapper since, without it, it would be difficult for users to do anything with the gaussians' means and covariance matrices.  However, I should have realized on the initial PR that we should define a new MultivariateGaussian class within spark.ml so that we do not reference spark.mllib publicly.  I can send a quick PR to add this on the Scala side.  @yanboliang does this sound reasonable to you?

For serialization to the Python side, we could use DataFrames if it simplifies the code.  To do that, we could keep a private reference to the SQLContext from training.